src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -2061,9 +2061,27 @@ void QgsPostgresConn::retrieveLayerTypes( QVector<QgsPostgresLayerProperty *> &l
       if ( sridAndTypeString == "NULL" )
         continue;
 
-      QStringList sridAndType = sridAndTypeString.split( ':' );
-      int srid = sridAndType[0].toInt();
+      const QStringList sridAndType = sridAndTypeString.split( ':' );
+      const int srid = sridAndType[0].toInt();
```

Comment:
given the complexity of the code that composes the sql statement, it would be prudent to add an assertion that sridAndType.size() == 3